### PR TITLE
[Cleanup][C-2] clean some `Program.random_seed` for test

### PR DESCRIPTION
**Pull request by enkilee**
### PR types

Others
### PR changes

Others
### Description

* https://github.com/PaddlePaddle/Paddle/issues/61385
C2
@gouzil

### CODE REVIEW

**Review by gouzil — DISMISSED** [review]
LGTM

File: test/legacy_test/dist_mnist.py [hunk 1]
```diff
@@ -27,8 +27,7 @@
 paddle.dataset.mnist.fetch()
 
 # Fix seed for test
-base.default_startup_program().random_seed = 1
-base.default_main_program().random_seed = 1
+paddle.seed(2023)
```
Comment (SigureMo) [inline test/legacy_test/dist_mnist.py]:
```suggestion
base.default_startup_program().random_seed = 1
base.default_main_program().random_seed = 1
```

**Review by SigureMo — COMMENTED** [review]
test_sync_batch_norm_op 会调用这个文件，先 revert

File: test/legacy_test/dist_mnist.py (hunk 1 above)
Comment (enkilee) [reply test/legacy_test/dist_mnist.py]: OK

File: test/legacy_test/dist_fleet_sync_batch_norm.py [hunk 2]
```diff
@@ -28,8 +28,7 @@
 
 def get_program(args):
     main, startup = Program(), Program()
-    main.random_seed = 10
-    startup.random_seed = 10
+    paddle.seed(2023)
```
Comment (SigureMo) [inline test/legacy_test/dist_fleet_sync_batch_norm.py]:
```suggestion
    main.random_seed = 10
    startup.random_seed = 10
```

**Review by SigureMo — APPROVED** [review]
LGTMeow

**Pull request merged**